bin/adhoc-scripts/updatePileupObjects.py

```diff
@@ -147,6 +158,65 @@ def writePileupDocs(mspileupUrl, puDocs, handler, logger):
                 logger.critical(msg)
 
 
+def addTransitionRecords(handler, url, userDN, logger, dryrun):
+    """
+    Helper function to add and update in place transition record for existing MSPileup records
+    :param handler: HTTP handler to use
+    :param url: url to use (string)
+    :param userDN: string representing user DN
+    :param logger: logger to use
+    :param dryrun: option to run dry-run mode (boolean)
+    :return: nothing
+    """
+    puDocs = getPileupDocs(url, handler, logger)
+    logger.info("Found %d documents in MSPileup", len(puDocs))
+    for rec in puDocs:
+        if not rec.get("transition"):
+            tranRecord = {'containerFraction': 1.0,
+                          'customDID': rec['pileupName'],
+                          'updateTime': gmtimeSeconds(),
+                          'DN': userDN}
+            rec['transition'] = [tranRecord]
+            logger.info("New pileup document is: %s", rec)
+
+    if dryrun:
+        logger.info("documents are not written due to --dry-run option\n")
+    else:
+        # finally, update the pileup documents in the database
+        writePileupDocs(url, puDocs, handler, logger)
+        logger.info("documents are uploaded to MSPileup\n")
+
+
+def updatePileupRecords(handler, url, fin, logger, dryrun):
+    """
+    helper function to update pileup records
+    :param handler: HTTP handler to use
+    :param url: url to use (string)
+    :param fin: input file name to read from (string)
+    :param logger: logger object to use
+    :param dryrun: option to run dry-run mode (boolean)
+    :return: nothing
+    """
+    with open(fin, 'r', encoding='utf-8') as istream:
```

Comment:
It looks like that with the current changes, we should stop making `fin` required in this script, no? In other words, I see 2 scenarios:
a) we provide a json file as input and write (update) them in the database
b) we do not provide a json file, fetch documents via MSPileup, update them in memory and persist them back in the database.

Reply:
I think both options are valid use-case and I adjusted the code to support them both.